node/utils.go

```diff
@@ -1,88 +1,216 @@
 package node
 
 import (
-	"bytes"
 	"context"
-	"encoding/binary"
 	"errors"
 	"fmt"
 
+	"github.com/Layr-Labs/eigenda/api"
 	pb "github.com/Layr-Labs/eigenda/api/grpc/node"
 	"github.com/Layr-Labs/eigenda/common/pubip"
 	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/consensys/gnark-crypto/ecc/bn254"
+	"github.com/consensys/gnark-crypto/ecc/bn254/fp"
+	"github.com/gammazero/workerpool"
 )
 
-const (
-	// Caution: the change to these prefixes needs to handle the backward compatibility,
-	// making sure the new code work with old data in DA Node store.
-	blobHeaderPrefix      = "_BLOB_HEADER_"  // The prefix of the blob header key.
-	batchHeaderPrefix     = "_BATCH_HEADER_" // The prefix of the batch header key.
-	batchExpirationPrefix = "_EXPIRATION_"   // The prefix of the batch expiration key.
-)
-
-// EncodeBlobKey returns an encoded key as blob identification.
-func EncodeBlobKey(batchHeaderHash [32]byte, blobIndex int, quorumID core.QuorumID) ([]byte, error) {
-	buf := bytes.NewBuffer(batchHeaderHash[:])
-	err := binary.Write(buf, binary.LittleEndian, int32(blobIndex))
-	if err != nil {
-		return nil, err
+// GetBatchHeader constructs a core.BatchHeader from a proto of pb.StoreChunksRequest.
+// Note the StoreChunksRequest is validated as soon as it enters the node gRPC
+// interface, see grpc.Server.validateStoreChunkRequest.
+func GetBatchHeader(in *pb.BatchHeader) (*core.BatchHeader, error) {
```

Comment:
moving a bunch of util methods from `node/grpc/utils.go` to prevent circular dependency